src/client/client.c

```diff
@@ -306,10 +187,15 @@ int client_call_manager(Client *client) {
                 r = print_client_usage("hirte");
         } else if (streq(client->op, "list-units")) {
                 if (client->opargc == 0) {
-                        r = method_list_units_on_all(client);
-                } else {
-                        r = method_list_units_on(client->opargv[0], client);
+                        r = method_list_units_on_all(
+                                        client->api_bus, print_unit_list_simple, client->opt_filter_glob);
```

Comment:
I understand that with `--filter`, the `hirtectl list-units node` functionality becomes redundant, it makes sense to remove it. But, on the other hand using it is much more intuitive and aligns with other commands, e.g. lifecycle. Maybe it will be not that bad to leave this option as well, although it basically does the same

Reply:
The implementation on the backend can stay the same, but we can just handle `hirtectl list-units node` by adding a filter

Reply:
> I understand that with `--filter`, the `hirtectl list-units node` functionality becomes redundant, it makes sense to
> remove it. But, on the other hand using it is much more intuitive and aligns with other commands, e.g. lifecycle.
> Maybe it will be not that bad to leave this option as well, although it basically does the same? WDYT?

Howdy @mkemel, I believe the commads are two different things:

- we cannot break the existing commands for users. (Okay, the hirte project is new but still) What happens for users using scripts based on hirtectl? They will break :)
- `hirtectl list-units` will list all nodes services instead of single one for example.

Let's compare:

** Only qm-node1 services output **
```
[root@control hirte]# ./builddir/src/client/hirtectl list-units qm-node1 | wc -l
122
```

** ALL 3 nodes reporting services **
```
[root@control hirte]# ./builddir/src/client/hirtectl list-units | wc -l
596
```

Now using a new flag from @engelmi :
```
# ./builddir/src/client/hirtectl list-units qm-node1 --filter=time-set.target
NODE                |ID                                                         |   ACTIVE|      SUB
====================================================================================================
qm-node1            |time-set.target                                            | inactive|     dead
```

Reply:
> I understand that with --filter, the hirtectl list-units node functionality becomes redundant, it makes sense to remove it. 

@mkemel I am not sure I understand which functionality you refer to that has been removed? It still works the same:
```bash
hirtectl list-units [node] --filter=xyz
```
In addition, the `--filter` option can be applied in both cases - when specifying a node and omitting it. However, I aligned the print output so that both print the node name, which seems more consistent.

Reply:
My bad, I have read the code incorrectly